polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -1862,8 +1865,9 @@ private FileIO loadFileIO(String ioImpl, Map<String, String> properties) {
     Map<String, String> propertiesWithS3CustomizedClientFactory = new HashMap<>(properties);
     propertiesWithS3CustomizedClientFactory.put(
         S3FileIOProperties.CLIENT_FACTORY, PolarisS3FileIOClientFactory.class.getName());
-    return CatalogUtil.loadFileIO(
-        ioImpl, propertiesWithS3CustomizedClientFactory, new Configuration());
+    return fileIOWrapperFactory.wrap(
+        CatalogUtil.loadFileIO(
+            ioImpl, propertiesWithS3CustomizedClientFactory, new Configuration()));
```

Comment:
Rather than defining a _wrapper_ factory, I think it makes sense to define a `FileIOFactory` that could have a wrapper impl and a default impl that simply calls `CatalogUtil.loadFileIO`.

Reply:
I'm curious what the FileIOFactory interface would look like in that case. In that situation I'm imagining 1 main method that would have to take the same arguments as `CatalogUtil.loadFileIO` so it can proxy them through. It would work but would be centered around `CatalogUtil.loadFileIO` almost always being used to construct the inner FileIO. I'm not super familiar with the Iceberg ecosystem so I don't know if that's the right decision, but having a concept of a "FileIOWrapper" removes the reliance on particular implementations of FileIO.

Reply:
I'd probably use the signature of our `loadFileIO` method, which is just
```java
loadFileIO(String ioImpl, Map<String, String> properties)
```
Personally, I think it's a bit annoying that we pass in this Hadoop `Configuration` object for no reason other than to satisfy the compiler. Just the class name and the properties is generic enough it supports _any_ FileIO implementation and doesn't explicitly require the caller to wrap an existing FileIO while still allowing for that use case.